test/e2e/operator_test.go

```diff
@@ -184,28 +183,97 @@ func TestMain(m *testing.M) {
+	// Get the number of master nodes
+	if err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+		klog.Infof("Listing nodes...")
+		nodeItems, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			klog.Errorf("Unable to list nodes: %v", err)
+			return false, nil
+		}
+		for _, node := range nodeItems.Items {
+			if _, exists := node.Labels["node-role.kubernetes.io/master"]; !exists {
+				continue
+			}
+			webhooksRunning[node.Name] = false
+		}
+		return true, nil
+	}); err != nil {
+		klog.Errorf("Unable to collect master nodes")
+		os.Exit(1)
+	}
+
+	// Wait until the secondary scheduler pod is running
+	if err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+		klog.Infof("Listing pods...")
+		podItems, err := kubeClient.CoreV1().Pods("run-once-duration-override").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			klog.Errorf("Unable to list pods: %v", err)
+			return false, nil
+		}
+		for _, pod := range podItems.Items {
+			if !strings.HasPrefix(pod.Name, "run-once-duration-override-") {
+				continue
+			}
+			klog.Infof("Checking pod: %v, phase: %v, deletionTS: %v\n", pod.Name, pod.Status.Phase, pod.GetDeletionTimestamp())
+			if pod.Status.Phase == corev1.PodRunning && pod.GetDeletionTimestamp() == nil {
+				webhooksRunning[pod.Spec.NodeName] = true
+			}
+		}
+
+		for node, running := range webhooksRunning {
+			if !running {
+				klog.Infof("Admission scheduled on node %q is not running", node)
+				return false, nil
+			}
+		}
+
+		return true, nil
+	}); err != nil {
+		klog.Errorf("Unable to wait for all webhook pods running: %v", err)
+		os.Exit(1)
+	}
+
+	klog.Infof("All daemonset webhooks are running")
+
 	os.Exit(m.Run())
 }
 
 func TestRunOnceDurationOverriding(t *testing.T) {
-	kubeClient := getKubeClientOrDie()
+	// runoncedurationoverrides.admission.apps.openshift.io/enabled: "true"
+	ctx, cancelFnc := context.WithCancel(context.TODO())
+	defer cancelFnc()
 
-	ctx := context.TODO()
+	clientSet := getKubeClientOrDie()
+
+	testNamespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "e2e-" + strings.ToLower(t.Name()),
+			Labels: map[string]string{
+				"runoncedurationoverrides.admission.apps.openshift.io/enabled": "true",
+			},
+		},
+	}
+	if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unable to create ns %v", testNamespace.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace.Name, metav1.DeleteOptions{})
 
 	pod := &corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: "default",
-			Name:      "test-run-once-duration-override-pod",
-			Labels:    map[string]string{"app": "test-run-once-duration-override"},
+			Namespace: testNamespace.Name,
+			Name:      "test-mutating-admission-pod",
 		},
 		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyOnFailure,
```

Comment:
Shouldn't we add RestartPolicy `Never` or `OnFailure` to expect `ActiveDeadlineSeconds` is set?. But in this test https://github.com/openshift/run-once-duration-override/blob/main/test/e2e/operator_test.go#L255, restartpolicy is not set and default is always.